lib/kafka_producer.js

```diff
@@ -202,8 +203,16 @@ KafkaProducer.prototype.batch = function batch(topic, message, timeStamp, callba
         return;
     }
 
+    if (typeof message !== 'string' && !Buffer.isBuffer(message)) {
+        var err = new Error('For batching, message must be a string or buffer, not ' + (typeof message));
+        if (!callback) {
+            throw err;
```

Comment:
Sounds good. I've changed both `kafka_producer.js` and `message_batch.js` to use `callback()` if provided, otherwise drop the message.